share/availability/discovery/discovery.go

```diff
@@ -247,7 +189,56 @@ func (d *Discovery) ensurePeers(ctx context.Context) {
 	}
 }
 
-func (d *Discovery) findPeers(ctx context.Context) bool {
+// disconnectsLoop listen for disconnect events and ensures Discovery state
+// is updated.
+func (d *Discovery) disconnectsLoop(ctx context.Context) {
+	// subscribe on EventBus in order to catch disconnected peers and restart
+	// the discovery. We specify a larger buffer size for the channel where
+	// EvtPeerConnectednessChanged events are sent (by default it is 16, we
+	// specify 32) to avoid any blocks on writing to the full channel.
+	sub, err := d.host.EventBus().Subscribe(&event.EvtPeerConnectednessChanged{}, eventbus.BufSize(eventbusBufSize))
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	defer sub.Close()
```

Comment:
Would be nice if you init subscribtion in Discovery constractor and provide it to runner routine. So if it error, we would catch it in FX. In current implementation there would be only error log line, but node would start normally and it is not good since this would break discovery.